### PR TITLE
e2e proposal edit mock structure

### LINKED ISSUE #2645: e2e: Proposal Edit mock structure.
Part of #2642.

Proposal Edit tests are still using the online api approach. We need to migrate it to the new mock structure introduced by (#2624).

We currently don't need new test suites, the main focus here is to first migrate and then code more tests using this new structure.

## Tips and Tricks

- Only mock requests related to Proposal Edit page
- Run only `edit` test file: `yarn cy:run -- --spec "cypress/e2e/proposal/edit.js"`
- Setup the API environment for all possible routes reached by the tests suite using a `beforeEach` statement
- Check on politeiawww's logs to check if all requests were mocked properly
- Use `proposal/list.js` tests as reference

### PR DESCRIPTION
**Pull request by vibros68**
Closes #2645

This diff implement mock data for edit proposal page. So the teste2e
will not submit to the server. This  new mock structure introduced by
[#2624](https://github.com/decred/politeiagui/pull/2624).

### CODE REVIEW

**Review by victorgcramos — CHANGES_REQUESTED** [review]
Hey mate, thanks for the PR. Noticed that your first test case hits the `/comments/v1/comments` on the backend. Also, left some inline comments to improve test readability and flow.

Notice that we already use middlewares for making request aliases, so you can use middleware's alias instead of `cy.route()`.

File: teste2e/cypress/support/core/generate.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ export function Record({
   const token = recordToken || Token();
   const status = recordStatus || faker.datatype.number(3) + 1;
   const state = recordState || faker.datatype.number(1) + 1;
-  const user = author instanceof User || new User();
+  const user = author || new User();
```
[the remark this answers is not in the document]
Comment (vibros68) [reply teste2e/cypress/support/core/generate.js:100]: `instanceof` work on the normal function. But after pass via cy middleware function, it is nolonger working

**Review by victorgcramos — APPROVED** [review]
LGTM! Just need to fix https://github.com/decred/politeiagui/pull/2692/files#r770595023 and we should be good to go! Good job!

Screen Shot 2021-12-20 at 12 46 05 PM

**Comment by vibros68** [comment]
Thank you @victorgcramos

File: teste2e/cypress/support/core/api.js [hunk 2]
```diff
@@ -110,6 +122,23 @@ export function detailsReply({
...
+    files,
+    author: username,
+    publickey,
+    signature,
+    token
+  });
+  return { record };
+}
+
+export function editRecordReply({
```
[the remark this answers is not in the document]
Comment (tiagoalvesdulce) [reply teste2e/cypress/support/core/api.js:129]: I think this comment wasn't addressed.

**Review by tiagoalvesdulce — APPROVED** [review]
LGTM! Left a small comment.

File: teste2e/cypress/support/core/api.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (vibros68) [reply teste2e/cypress/support/core/api.js:129]: It is addressed. If we do not pass the `version` param. The function will generate a randomly greater than 1 value. The code is here: 
`const version = recordVersion || faker.datatype.number(6) + 1;`

[the remark this answers is not in the document]
Comment (tiagoalvesdulce) [reply teste2e/cypress/support/core/api.js:129]: oh, my bad

**Pull request merged**